### PR TITLE
Improve ModulePath calculation and refactor related method names

### PR DESCRIPTION
**Pull request by daxian-dbw**
Resolving #1744

The PR include following changes:
1. Refactor method names related to module path calculation.
2. Improve the logic that detects if a module path is PSHome module path derived from a different powershell core instance.
3. Add more sxs module path tests.

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@mirichmo could you please review this PR?

File: test/powershell/enginecore/ModulePath.Tests.ps1 [hunk 1]
```diff
@@ -38,8 +48,44 @@ Describe "SxS Module Path Basic Tests" -tags "CI" {
...
+        $paths[2].TrimEnd([System.IO.Path]::DirectorySeparatorChar) | Should Be $expectedSystemPath
+    }
+
+
+    It "ignore pshome module path derived from a different powershell core instance" -Skip:(!$IsCoreCLR) {
+
+        ## Create 'powershell' and 'powershell.deps.json' in the fake PSHome folder,
+        ## so that the module path calculation logic would believe it's real.
+        New-Item -Path $fakePowerShell -ItemType File -Force > $null
+        New-Item -Path $fakePSDepsFile -ItemType File -Force > $null
```
Comment (mirichmo) [inline test/powershell/enginecore/ModulePath.Tests.ps1]: The creation of the fake powershell.exe and deps file should be done in a BeforeEach and After each for this test. It should not be cleaned up in the next test. I suggest using a Context to limit the BeforeEach and AfterEach to this test.

File: test/powershell/enginecore/ModulePath.Tests.ps1 [hunk 2]
```diff
@@ -38,8 +48,44 @@ Describe "SxS Module Path Basic Tests" -tags "CI" {
...
+        $paths[1].TrimEnd([System.IO.Path]::DirectorySeparatorChar) | Should Be $expectedSharedPath
+        $paths[2].TrimEnd([System.IO.Path]::DirectorySeparatorChar) | Should Be $expectedSystemPath
+    }
+
+
+    It "ignore pshome module path derived from a different powershell core instance" -Skip:(!$IsCoreCLR) {
+
+        ## Create 'powershell' and 'powershell.deps.json' in the fake PSHome folder,
+        ## so that the module path calculation logic would believe it's real.
+        New-Item -Path $fakePowerShell -ItemType File -Force > $null
```
Comment (mirichmo) [inline test/powershell/enginecore/ModulePath.Tests.ps1]: The creation of the fake powershell.exe and deps file should be done in a BeforeEach and After each for this test. It should not be cleaned up in the next test. I suggest using a Context to limit the BeforeEach and AfterEach to this test.

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 3]
```diff
@@ -2823,7 +2823,7 @@ internal Exception BindRunspace(Runspace initializedRunspace, PSTraceSource runs
             // If a user has any module with the same name as that of the core module( or nested module inside the core module)
             // in his module path, then that will get loaded instead of the actual nested module (from the GAC - in our case)
```
Comment (mirichmo) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Are all 4 commits necessary for the history, or can they be squashed?

**Review by mirichmo — CHANGES_REQUESTED** [review]
The code changes look good. I have one issue with the tests though.

File: test/powershell/enginecore/ModulePath.Tests.ps1 (hunk 2 above)
Comment (daxian-dbw) [inline test/powershell/enginecore/ModulePath.Tests.ps1]: Will fix it.

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 3 above)
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: I think we should keep them separate commits, especially the one that rename the method names.

Comment (daxian-dbw) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: I have squashed test changes into one commit.

**Review by mirichmo — APPROVED** [review]
LGTM

**Pull request merged**